ed25519/src/ed25519_verify.rs

```diff
@@ -0,0 +1,46 @@
+use ed25519_dalek::{Signature,PublicKey};
+use sha2::Sha512;
+
+
+#[derive(Debug)]
+pub struct Error(pub &'static str);
+
+impl From<&'static str> for self::Error {
+  fn from(val: &'static str) -> Self {
+    self::Error(val)
+  }
+}
+
+pub fn verify(input: &[u8; 128]) -> Result<&[u8; 4], self::Error> {
```

Comment:
Why does this work..? This seems like an obvious dangling reference.

Reply:
Okay it only works when the slice being returned is a constant array 'literal'.  But if I omit the parameter, I get an error complaining about the lifetime of the returned slice...